src/types.ts

```diff
@@ -76,17 +70,24 @@ interface BaseLoggerOptions extends pino.LoggerOptions {}
 export type Logger = pino.Logger & BaseLoggerOptions;
 
 export type InferContext<T> = T extends Elysia<
-  infer Path,
+  infer _Path,
   infer _Scoped,
   infer Singleton,
   infer _Definitions,
-  infer _Metadata,
-  infer Routes,
-  infer _EphemeralMetadata
+  infer Metadata,
+  infer _Routes,
+  infer Ephemeral,
+  infer Volatile
 >
-  ? Context<Routes, SingletonBase, Path> &
-      Partial<Singleton["derive"]> &
-      Partial<Singleton["decorator"]>
+  ? Partial<
+      Context<
+        Metadata["schema"] & Ephemeral["schema"] & Volatile["schema"],
```

Comment:
These changes are not necessary (at least on my side) to get correct types in the provided sample code: https://github.com/bogeychan/elysia-logger/issues/14#issuecomment-2016045792

Can you please share some code that justifies the changes suggested here?

Reply:
something like this, the new Elysis's generic is different for the derive type, so you will get a type error when you cast the type of context with InferContext in CustomProps or autoLogging

```
const app = new Elysia()
  .derive(() => ({
    foo: 'bar',
  }));

app
  .use(
    logger({
      customProps({ foo }: InferContext<typeof main>) {
        return {
          foo,
        };
      },
      autoLogging: true,
    })
  )
```